### ISSUE TITLE
Checkstyle-IDEA shows full stack trace of parse error

### ISSUE THREAD

**Issue by kevin-wayne** [jshiell/checkstyle-idea#409]
When a Java program can't be parsed, Checkstyle-IDEA displays a full Java stack trace of the underlying parse error (see below) when you hover over the error stripe. While this is consistent with what Checkstyle would display when run from the command line, it would be much nicer if Checkstyle-IDEA caught the CheckstyleException/MismatchedTokenException and reported a more informative message (e.g., that the code could not be parsed, presumably because it is not a valid Java program).

For context, I'm teaching an intro programming class and the resulting error message suggests (to some novice programmers) a bug in Checkstyle-IDEA or IntelliJ (when, in fact, it's simply that their Java program couldn't be parsed by Checkstyle/antlr).
```
Checkstyle: Got an exception - com.puppycrawl.tools.checkstyle.api.CheckstyleException: MismatchedTokenException occurred while parsing file /var/folders/ht/36vy3rgd5kvczk6_0d73fqhr0000gn/T/csi-281/Percolation.java. at com.puppycrawl.tools.checkstyle.JavaParser.parse(JavaParser.java:98) at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:180) at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:81) at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:316) at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:286) at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:213) at org.infernus.idea.checkstyle.service.cmd.OpScan.processAndAudit(OpScan.java:75) at org.infernus.idea.checkstyle.service.cmd.OpScan.execute(OpScan.java:52) at org.infernus.idea.checkstyle.service.cmd.OpScan.execute(OpScan.java:22) at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.executeCommand(CheckstyleActionsImpl.java:130) at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.scan(CheckstyleActionsImpl.java:76) at org.infernus.idea.checkstyle.checker.CheckStyleChecker.scan(CheckStyleChecker.java:45) at org.infernus.idea.checkstyle.CheckStyleInspection.lambda$inspectFile$1(CheckStyleInspection.java:91) at java.util.Optional.map(Optional.java:215) at org.infernus.idea.checkstyle.CheckStyleInspection.inspectFile(CheckStyleInspection.java:91) at org.infernus.idea.checkstyle.CheckStyleInspection.lambda$checkFile$0(CheckStyleInspection.java:63) at com.intellij.openapi.application.impl.ApplicationImpl$2.call(ApplicationImpl.java:337) at java.util.concurrent.FutureTask.run(FutureTask.java:266) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) at java.lang.Thread.run(Thread.java:745) Caused by: /var/folders/ht/36vy3rgd5kvczk6_0d73fqhr0000gn/T/csi-281/Percolation.java:39:9: expecting EOF, found 'grid' at antlr.Parser.match(Parser.java:211) at com.puppycrawl.tools.checkstyle.grammars.GeneratedJavaRecognizer.compilationUnit(GeneratedJavaRecognizer.java:211) at com.puppycrawl.tools.checkstyle.JavaParser.parse(JavaParser.java:92) ... 20 more more... (⌘F1)
```

**Comment by jshiell** [comment]
Thanks for the suggestion - I think it's a great one.

This has always been a bit of a spotty field, as the Checkstyle devs are very upfront in that the tool is designed for CLI use on static files, and using it as an API for real-time tools is not a supported path. Hence these errors have always been outside their scope for addressing - which is not to say we shouldn't make the effort.

I'm busy this weekend but will try and have a look at this in the not too distant.

**Comment by kevin-wayne** [comment]
Thanks, would be great to have that feature when you have spare time.

**Comment by jshiell** [comment]
Released in 5.22.1.

I could find a path for the inline scanning - there's a case especially to hide them. But I have improved the static scan and the event log output.

If you spot any other paths, please open them with examples and I'll try get them fixed as well. Thank you.
